### PR TITLE
import vsts build data for test dashboard

### PR DESCRIPTION
**Pull request by philipktlin**
import vsts builds from branches other than master to test dashboard database.

### CODE REVIEW

File: tools/IoTEdgeDevOps/VstsPipelineSync/VstsBuildBatchUpdate.cs [hunk 1]
```diff
@@ -35,9 +38,12 @@ public async Task RunAsync(TimeSpan waitPeriodAfterEachUpdate, CancellationToken
...
                     Console.WriteLine($"Import Vsts Builds data started at {DateTime.UtcNow}");
 
-                    lastUpdatePerBranchPerDefinition.Upsert(
-                        "refs/heads/master",
-                        await ImportVstsBuildsDataAsync(buildManagement, "refs/heads/master", BuildExtension.MasterBranchBuildDefinitions));
+                    foreach (string branch in this.branches)
+                    {
+                        lastUpdatePerBranchPerDefinition.Upsert(
+                            branch, 
+                            await ImportVstsBuildsDataAsync(buildManagement, branch, BuildExtension.MasterBranchBuildDefinitions));
```
Comment (and-rewsmith) [inline tools/IoTEdgeDevOps/VstsPipelineSync/VstsBuildBatchUpdate.cs]: Recommend updating the name ```MasterBranchBuildDefinitions```

File: tools/IoTEdgeDevOps/VstsPipelineSync/VstsBuildBatchUpdate.cs [hunk 2]
```diff
@@ -15,14 +15,17 @@ public class VstsBuildBatchUpdate
         readonly DevOpsAccessSetting devOpsAccessSetting;
         readonly string dbConnectionString;
         Dictionary<string, Dictionary<BuildDefinitionId, DateTime>> lastUpdatePerBranchPerDefinition;
+        HashSet<string> branches;
 
-        public VstsBuildBatchUpdate(DevOpsAccessSetting devOpsAccessSetting, string dbConnectionString)
+        public VstsBuildBatchUpdate(DevOpsAccessSetting devOpsAccessSetting, string dbConnectionString, HashSet<string> branches)
         {
             ValidationUtil.ThrowIfNull(devOpsAccessSetting, nameof(devOpsAccessSetting));
+            ValidationUtil.ThrowIfNulOrEmptySet(branches, nameof(branches));
```
Comment (and-rewsmith) [inline tools/IoTEdgeDevOps/VstsPipelineSync/VstsBuildBatchUpdate.cs]: Typo in ```Nul```?

Also, I think we should do a test run to make sure this works.

File: tools/IoTEdgeDevOps/VstsPipelineSync/VstsBuildBatchUpdate.cs [hunk 3]
```diff
@@ -15,14 +15,17 @@ public class VstsBuildBatchUpdate
         readonly DevOpsAccessSetting devOpsAccessSetting;
         readonly string dbConnectionString;
         Dictionary<string, Dictionary<BuildDefinitionId, DateTime>> lastUpdatePerBranchPerDefinition;
+        HashSet<string> branches;
 
-        public VstsBuildBatchUpdate(DevOpsAccessSetting devOpsAccessSetting, string dbConnectionString)
+        public VstsBuildBatchUpdate(DevOpsAccessSetting devOpsAccessSetting, string dbConnectionString, HashSet<string> branches)
         {
             ValidationUtil.ThrowIfNull(devOpsAccessSetting, nameof(devOpsAccessSetting));
```
Comment (and-rewsmith) [inline tools/IoTEdgeDevOps/VstsPipelineSync/VstsBuildBatchUpdate.cs]: Why can't we use the Preconditions from Edge.Util?

Comment (philipktlin) [reply tools/IoTEdgeDevOps/VstsPipelineSync/VstsBuildBatchUpdate.cs]: I just don't want to have dependency from edge.util.  If someone changed anything in edge.util, they don't know that may break anything in devops solution.

File: tools/IoTEdgeDevOps/VstsPipelineSync/VstsBuildBatchUpdate.cs (hunk 2 above)
Comment (philipktlin) [reply tools/IoTEdgeDevOps/VstsPipelineSync/VstsBuildBatchUpdate.cs]: I ran the solution; just the method name has typo.

**Pull request merged**